### PR TITLE
Logging Manager

### PR DESCRIPTION
**Pull request by JackRiga**
# Logging Manager
## Description
- In this PR I have added a logging manager which when called will output messages to the UI within unity. 
- This allows us to log more readable messages for ourselves while debugging as well as messages for the player of the game
- - E.g. Telling the player that they can't place a building for X reason

## Why was this needed?
- We had nothing in place to convey any information to the player
- The debug log is small and annoying to read for developers
- It would be nice to keep the debug log purely for when exceptions occur.
- - Saves us from having to wade through many unrelated log messages

## Configuration 
- N/A

### CODE REVIEW

File: Assets/_Scripts/3Managers/Logger.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+public class Logger : MonoBehaviour
+{
+    [SerializeField]
+    private GameObject LogPrefab;
+
+    public enum LogType
+    {
+        Error,
+        Debug,
```
Comment (Hindsight420) [inline Assets/_Scripts/3Managers/Logger.cs:15]: What's debug for? I would expect: Error, Warning, Info/Message

File: Assets/_Scripts/3Managers/IslandManager.cs [hunk 2]
```diff
@@ -11,6 +11,7 @@ public class IslandManager : Singleton<IslandManager>
 
     public Island Island;
     [SerializeField] IslandView view;
+    [SerializeField] Logger Logger;
```
Comment (Hindsight420) [inline Assets/_Scripts/3Managers/IslandManager.cs]: Can we make Logger static? Or maybe a singleton? I'd rather not initialize this through the inspector, especially if we want to use this in other managers as well.

File: Assets/_Scripts/3Managers/IslandManager.cs [hunk 3]
```diff
@@ -36,10 +37,10 @@ public void TryPlaceBuilding(Tile tile, BuildingType buildingType)
         //TODO: This needs to be moved at a later point.
         var validator = buildingType.GetBuildingValidator(tile);
         var buildingErrors = validator.ValidateResources(buildingType);
-        if (buildingErrors.Any()) { buildingErrors.ForEach(x => Debug.Log(x)); return; }
+        if (buildingErrors.Any()) { buildingErrors.ForEach(x => Logger.LogMessage(x, Logger.LogType.Error)); return; }
```
Comment (Hindsight420) [inline Assets/_Scripts/3Managers/IslandManager.cs]: The logger has no good support for multiple error messages, showing them all in the same position:
image

**Review by Hindsight420 — COMMENTED** [review]
The Logging object is always showing as a grey panel, is that intended?
image

File: Assets/_Scripts/3Managers/Logger.cs (hunk 1 above)
Comment (JackRiga) [reply Assets/_Scripts/3Managers/Logger.cs:15]: For us

File: Assets/_Scripts/3Managers/IslandManager.cs (hunk 2 above)
Comment (Hindsight420) [reply Assets/_Scripts/3Managers/IslandManager.cs]: @JackRiga

**Review by Hindsight420 — APPROVED** [review]

**Pull request merged**